runtime/src/accounts_db.rs

```diff
@@ -7905,27 +7893,18 @@ impl AccountsDb {
 
         let mut measure = Measure::start("shrink");
         let mut shrink_candidate_slots = self.shrink_candidate_slots.lock().unwrap();
-        for (slot, slot_shrink_candidates) in new_shrink_candidates {
-            for (store_id, store) in slot_shrink_candidates {
-                // count could be == 0 if multiple accounts are removed
-                // at once
-                if store.count() != 0 {
-                    debug!(
-                        "adding: {} {} to shrink candidates: count: {}/{} bytes: {}/{}",
-                        store_id,
-                        slot,
-                        store.approx_stored_count(),
-                        store.count(),
-                        store.alive_bytes(),
-                        store.total_bytes()
-                    );
+        for (slot, store) in new_shrink_candidates {
+            debug!(
+                "adding: {} {} to shrink candidates: count: {}/{} bytes: {}/{}",
+                store.append_vec_id(),
+                slot,
+                store.approx_stored_count(),
+                store.count(),
+                store.alive_bytes(),
+                store.total_bytes()
+            );
 
-                    shrink_candidate_slots
-                        .entry(slot)
-                        .or_default()
-                        .insert(store_id, store);
-                }
-            }
+            shrink_candidate_slots.insert(slot, store);
```

Comment:
I think the store.count() != 0 condition is still valid.

Reply:
There will be a store or not. `store` is not an option in this data structure. It used to be a vec. A vec can be empty and have 0 elements, which we can skip. But, a single store can never be missing (ie. 0).